addons/mail/models/mail_message.py

```diff
@@ -176,7 +170,7 @@ def mark_all_as_read(self, channel_ids=None, domain=None):
                 ('res_partner_id', '=', self.env.user.partner_id.id),
                 ('is_read', '=', False)])
             if delete_mode:
-                notifications.unlink()
+                notifications.write({'is_read': True, 'active': False})
```

Comment:
Is there any reason to consider differently notifications for share users and notifications for classic users ?